src/s3path.jl

```diff
@@ -307,7 +313,9 @@ function Base.mkdir(fp::S3Path; recursive=false, exist_ok=false)
     else
         if hasparent(fp) && !exists(parent(fp))
             if recursive
-                mkdir(parent(fp); recursive=recursive, exist_ok=exist_ok)
+                # don't try to create buckets this way, minio at least really doesn't like it
```

Comment:
yeah, I think it's totally fine to not allow `mkdir` to make a bucket, that would be surprising to me if it happened. It's not like `mkdir` makes a new drive on windows or something if you do `mkdir("E://")`.

Reply:
For what it's worth, previously it just seemed to be downright broken in certain situations.  And yeah, creating the bucket with `makedir` seemed a little excessive.  Even more so since we've by now discussed extensively, `mkdir` and the functionality it makes use of on the S3 side are a bit of a hack.